### DISCUSSION TITLE
How to filter through folders registry

### DISCUSSION THREAD

**Discussion by mirek163** [renovatebot/renovate#30745] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Github version: latest

### Please tell us more about your question or problem

Hello,
I have question. Is it possible, to set in matchHost when going through artifactory , to not search SNAPSHOT directories  in registry? Basically all I just want to is that Renovate would not even try to look on these version. 

What I understanded Package rules would not help me much with this.
### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Try configuring an allowedVersions regex pattern

**Reply by mirek163** [reply]
But this i would need to make package rule and it would not apply on searching through those directories in registry but just filter final found, wouldnt it ?
I tried something like this, which didnt solve the problem either.

                        env.RENOVATE_PACKAGE_RULES = """[{
                        matchFileNames: ["**/*-SNAPSHOT/**"],
                        enabled: false
                        }]"""
and also this:

                          env.RENOVATE_PACKAGE_RULES = """[{
                          matchPackageNames:["*"]
                          allowedVersions: "!/^\\\\d+\\\\.\\\\d+-SNAPSHOT\$/"
                          }]"""

So did u meant using allowedVersions inside hostRules?

**Reply by rarkins** [reply]
Filtering will be applied after lookup. It's only valid in packageRules

**Reply by mirek163** [reply]
So there is no way to be applied when lookup?
Thanks for quick response.

**Reply by rarkins** [reply, marked as answer]
No way
